KLSadd_insertion/src/updateChapters.py

```diff
@@ -164,185 +400,237 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
                 word1 = str(p[count])
-                if (designator in word1[word1.find(
-                        "\\subsection*{") + 1: word1.find("}")]):
-                    chap[i - 2] += "%Begin KLSadd additions"
+                if designator in word1[word1.find("\\subsection*{") + 1: word1.find("}")]:
+                    chap[i - 2] += "%Begin KLS Addendum additions"
                     chap[i - 2] += p[count]
-                    chap[i - 2] += "%End of KLSadd additions"
+                    chap[i - 2] += "%End of KLS Addendum additions"
                     count += 1
                 else:
                     count += 1
+    return chap
 
 
-def referencePlacerII(chap, references, p, kls, refII, refIII, specref):
-    if chapticker2 == 0:
-        designator = "9."
-    elif chapticker2 == 1:
-        designator = "14."
-    count = 0
-    sectionnuma = specref[0]
-    sectionnumb = specref[1]
-    for i in specref:
-        pass
+# method to change file string(actually a list right now), returns string to be written to file
+# If you write a method that changes something, it is preffered that you call the method in here
+def fix_chapter(chap, references, paragraphs_to_be_added, kls, kls_list_all, chapticker2, new_commands, klsaddparas, sortmatch_2, tempref, sorter_check):
+    """
+    Removes specific lines stopping the latex file from converting into python, as well as running the
+    functions responsible for sorting sections and placing the correct additions in the correct places
 
+    :param chap: Chapter being processed.
+    :param references: List containing references (used in reference placer).
+    :param paragraphs_to_be_added: List containing additions to be added.
+    :param kls: The addendum that contains sections to be added (not processed).
+    :param kls_list_all: List of all identified keywords, fed into the chapter sorter.
+    :param chapticker2: Which chapter is being searched (9 or 14).
+    :return:
+    """
 
-# method to change file string(actually a list right now), returns string to be written to file
-# If you write a method that changes something, it is preffered that you
-# call the method in here
-def fixChapter(chap, references, p, kls, refII, refIII, specref):
-    # chap is the file string(actually a list), references is the specific references for the file,
-    # and p is the paras variable(not sure if needed) kls is the KLSadd.tex as
-    # a list
-    referencePlacer(chap, references, p, kls, refII, refIII, specref)
-    chap = prepareForPDF(chap)
-    cms = getCommands(kls)
-    chap = insertCommands(kls, chap, cms)
+    sort_location = 0
+
+    for i in kls_list_all:
+        fix_chapter_sort(kls, chap, i, sort_location, klsaddparas, sortmatch_2, tempref, sorter_check)
+        sort_location += 1
+
+    for a in range(len(paragraphs_to_be_added)-1):
+        for b in sortmatch_2:
+            for c in b:
```

Comment:
All of these variable names should be changed to something more self documenting than letters